Add descriptions for plugin names

djyasin opened the pull request:
##### SUMMARY
<!--- Describe the change, including rationale and design decisions -->
This PR is to add descriptions to inventory plugin names
<!---
If you are fixing an existing issue, please include "related #nnn" in your
commit message and your description; but you should still explain what
the change does.
-->

##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->
 - Bug, Docs Fix or other nominal change

##### COMPONENT NAME
<!--- Name of the module/plugin/module/task -->
 - API


##### AWX VERSION
<!--- Paste verbatim output from `make VERSION` between quotes below -->
```
awx: 24.6.2
```


##### ADDITIONAL INFORMATION
<!---
Include additional information to help people understand the change here.
For bugs that don't have a linked bug report, a step-by-step reproduction
of the problem is helpful.
  -->

<!--- Paste verbatim command output below, e.g. before and after your change -->
```

```

djyasin committed:
Add descriptions for plugin names

djyasin committed:
Update serializers to display plugin and plugin description

Remove hard coded descriptions and change for loop logic to pull them from awx-plugins

djyasin commented:
To test these changes, visit `/api/v2/inventory_sources/` and click `options`. You should now see both plugin names and their respective descriptions.

chrismeyersfsu commented on awx/main/utils/plugins.py:
```diff
@@ -56,4 +56,31 @@ def load_combined_inventory_source_options() -> dict[str, str]:
 
-    return dict(zip(plugins, plugins), file='file')
+    plugin_descriptions = {
+        'azure_rm': 'Microsoft Azure Resource Manager',
```
The description lives in `awx-plugins`. For example here is the azure description https://github.com/ansible/awx-plugins/blob/devel/src/awx_plugins/inventory/plugins.py#L154

We don't want the descriptions to be hard-coded here in awx because we don't know the entire set of plugins.

Say we introduce a new plugin. Call it `foo`. We wouldn't want to update this dictionary and add the description for `foo`. Instead, we want the description for `foo` to live _with_ the plugin, over in `awx-plugins`.

djyasin replied:
That makes 100% sense! I was missing a few process steps here that I didn't understand. I now see how this will be consumed.

chrismeyersfsu commented on awx/main/utils/plugins.py:
```diff
@@ -56,4 +56,31 @@ def load_combined_inventory_source_options() -> dict[str, str]:
+        'constructed': 'Template additional groups and hostvars at runtime',
+    }
+
+    result = {plugin: plugin_descriptions.get(plugin, plugin) for plugin in plugins}
```
Try changing this to:

`result = {plugin: plugin.plugin_description for plugin in plugins}`

djyasin replied:
Thank you so much! I have made that change and I now see how this is connecting. Thank you!

djyasin committed:
Add function to extract plugin name descriptions

djyasin committed:
Add description for scm

Changed double quotes to single quotes on scm description to see if that impacts the failing api schema test

Move descriptions manually added together for easier readability

chrismeyersfsu commented on awx/main/utils/plugins.py:
```diff
@@ -56,4 +74,12 @@ def load_combined_inventory_source_options() -> dict[str, str]:
-    return dict(zip(plugins, plugins), file='file')
+    plugin_description = discover_available_cloud_provider_descriptions()
+
+    plugin_description['scm'] = 'Sourced from a Project'
```
Since this is a plugin we don't want to assume that the `ssh` plugin will be discovered. Change this to something like:

```
if 'scm' in plugin_description:
    plugin_description['scm'] = 'Sourced from a Project'
```

djyasin replied:
@chrismeyersfsu since we are manually adding scm and constructed in `compute_cloud_inventory_sources`  those two are always present- Does that need to change? Maybe I'm thinking about that incorrectly.

chrismeyersfsu replied:
Thanks for pointing this out. You are correct. Marking this comment resolved

chrismeyersfsu commented on awx/main/utils/plugins.py:
```diff
@@ -56,4 +74,12 @@ def load_combined_inventory_source_options() -> dict[str, str]:
+
+    plugin_description['scm'] = 'Sourced from a Project'
+
+    plugin_description['file'] = 'File-based inventory source'
```
same as above

djyasin committed:
Conditionalize scm and file descriptions
